SCClassLibrary/Common/Control/Server.sc

```diff
@@ -970,24 +975,141 @@ Server {
 					// not sure this should be here or elsewhere...
 					// this.bootInit(recover);
 					if(startAliveThread) { statusWatcher.start };
+					appBooted = true
 				};
 			} {
 				statusWatcher.serverBooting = false;
 				"%.boot - server process rebooting.\n".postf(this);
-				this.reboot;
+				fork { 0.01.wait; this.reboot };
 			}
 		}, {
 			this.bootServerApp({
 				if (Server.postingBootInfo) {
 					"%.boot - after bootServerApp...\n".postf(this);
 				};
-				if(startAliveThread) { statusWatcher.start }
-			}, )
+				if(startAliveThread) { statusWatcher.start };
+			})
 		}, 0.25);
+
+		// quick inlining of initial bootsteps
+		fork {
+			var test, tests = { [this.applicationRunning, this.hasBooted, this.notified] };
+			while {
+				test = tests.().every(_ == true);
+				test.not and: { timeout > 0 }
+			} {
+				timeout = timeout - 0.1;
+				0.1.wait;
+			};
+			if (test) {
+				this.prRunBootTask;
+			} {
+				"% boot failed after timeout!".format(this).warn;
+			}
+		}
+	}
+
+	bootAlt { | startAliveThread = true, recover = false, onFailure, onComplete, timeout = 5 |
+		// measure time for all boot steps:
+		var t0 = thisThread.seconds, now = { thisThread.seconds - t0 };
+		var pt = { |str=""| "t %: %".format(now.().round(0.01), str).postln; };
```

Comment:
Everything should be made as short as possible, but not shorter, otherwise one day you'll hurt yourself with occam's razor.

Reply:
ok, I was in a hurry ;-) better like this?
```
		// measure time for all boot steps:
		var timeAtBootStart = thisThread.seconds, now = { thisThread.seconds - timeAtBootStart };
		var postTime = { |str=""| "time %: %".format(now.().round(0.01), str).postln; };
		var postTimeIfPosting = { |str=""| if (Server.postingBootInfo) { postTime.(str) } };
```